ui/src/data/actions/index.ts

```diff
@@ -0,0 +1,18 @@
+export type Action =
+  | ReturnType<typeof resetCachedQueryResults>
+  | ReturnType<typeof setQueryResultsByQueryID>
+
+export const hashCode = s =>
+  s.split('').reduce((a, b) => ((a << 5) - a + b.charCodeAt(0)) | 0, 0)
+
+export const setQueryResultsByQueryID = (queryID: string, files: string[]) =>
```

Comment:
it's really confusing when the variable is called `queryID` but you are actually passing in the queryText, and you will convert it into the ID inside the function. Could you change the name of this function to setQueryResultsForQuery(queryText: string, files: string[]))

Reply:
Definitely. Great point